src/Licensing/Licensing.ts

```diff
@@ -0,0 +1,46 @@
+import { ConnectionManager } from '../Connection/ConnectionManager';
+import { LicenseChangeResponse, LicenseStateResponse } from '../Connection/TouchFreeServiceTypes';
+
+export enum LicenseState {
+    UNLICENSED,
+    LICENSED,
+    CAMERA_UNLICENSED,
+}
+
+// Class: LicenseManager
+// This class contains a collection of static methods for interacting with the Licensing state of TouchFree Service.
+export class LicenseManager {
+    // Property: currentState
+    // Represents the current Licensing State of the service. May be out of date upon first
+    // connection until License state changes, or <GetLicenseState> is invoked at lease once.
+    public static currentState: LicenseState = LicenseState.UNLICENSED;
+
+    // Function: GetLicenseState
+    // Used to request a <LicenseState> representing the current state of the Service's Licenses via the WebSocket.
+    // Provides a <LicenseState> asynchronously via the _callback parameter.
+    //
+    // If your _callback requires context it should be bound to that context via .bind()
```

Comment:
💬  This line isn't true if using an arrow function (which implicitly binds). It's also a JS feature rather than our tooling feature so I don't think we want / need this line? (same for all instances of it)

Reply:
I think because of the way that we invoke the callbacks, that the arrow funcs don't inherently bind to the context, as at time of invocation they aren't actually arrow functions? That's a hazy memory from a while ago, but I remember the callbacks causing some weirdness there. If you can find a way to use one of these callbacks that binds context correctly with arrows I'm happy to remove these.

(The line is there even thought it's a JS thing because it caught people out because we had a few engagements with people who weren't super JS-ey and this is a kinda niche JS thing that will affect them if they don't know it)

Reply:
I can investigate on Monday (out of curiosity) 👍 
If people have run into in the past then happy for it to stay